src/playsim/p_enemy.cpp

```diff
@@ -2370,6 +2371,20 @@ DEFINE_ACTION_FUNCTION(AActor, A_Look2)
 	return 0;
 }
 
+void GoalReached(AActor* self, AActor* oldgoal)
+{
+	if (self && self->Level)
+		self->Level->localEventManager->WorldThingGoalReached(self, oldgoal);
+}
+
+DEFINE_ACTION_FUNCTION(AActor, GoalReached)
```

Comment:
This should be native
`DEFINE_ACTION_FUNCTION_NATIVE(AActor, GoalReached, GoalReached)`